lib/OA/Email.php

```diff
@@ -600,12 +600,17 @@ function sendCampaignImpendingExpiryEmail($oDate, $campaignId) {
                             // was crossed to the point where it's about to expire,
                             // so send those emails, baby!
                             foreach ($aUsers as $aUser) {
+                                Language_Loader::load('default', $aUser['language']);
+                                $user_date_format = $date_format;
+                                if ($GLOBALS['date_format']) {
```

Comment:
`$date_format` is already defined as global in https://github.com/revive-adserver/revive-adserver/blob/badef024f873608b4125a32a44270b7bc6cf71e5/lib/OA/Email.php#L436

Just `Language_Loader::load('default', $aUser['language']);` should be enough, as it is overwriting the variable in the global scope.

Reply:
However it might be best to restore the previous language after the foreach loop, to ensure the rest of the maintenenance tasks work as expected.